src/server/services/wms/qgswmsrestorer.h

```diff
@@ -56,6 +57,7 @@ class QgsLayerRestorer
     {
       QString name;
       double mOpacity;
+      QgsAbstractVectorLayerLabeling *mLabeling = nullptr;
```

Comment:
unique ptr?

Reply:
This is tricky, because the settings struct is copied. 

Technically the settings never owns the resource, it just keeps it for the duration of its lifetime and transfers it back to the layer when the destructor is called.

Also , the struct is private and it's not exposed to API,  given the (non-copyable) complications of turning this into a unique_ptr  I'd rather not follow your advice.

Reply:
BTW here is an attempt: https://github.com/qgis/QGIS/pull/53438/commits/ae23847c5998208d22343ef89bb0b8dbbaa8fe14